### ISSUE TITLE
Two additional models recommend for builting into rTPC. 

### ISSUE THREAD

**Issue by Jinta0Li** [padpadpadpad/rTPC#54]
Dear authors, 

I really like your packages, and have used it to analysis my data.

I would like to recommend two models, which both are derived from transition state theory. Perhaps you could consider building them into your rTPC.

1. The first one is the so-called MMRT (Macromolecular Rate Theory), which has recently been increasingly used in Ecology.

MMRT was firstly introduced by Hobbs et al. (2013). And Liang et al (2017) proved that the MMRT is formally equivalent to a second-order log-polynomial model (ie, ln y = a + b×T + c×T^2). More recently, it appears that a modified version of MMRT has been developed (Prentice et al, 2020; Alster et al, 2023).
_Refs_:
  Hobbs, et al. Change in Heat Capacity for Enzyme Catalysis Determines Temperature Dependence of Enzyme Catalyzed Rates. ACS Chem. Biol. 2013, 8, 11, 2388–2393. https://doi.org/10.1021/cb4005029
  Liang, et al. Macromolecular rate theory (MMRT) provides a thermodynamics rationale to underpin the convergent temperature response in plant leaf respiration. Glob Change Biol. 2018; 24: 1538–1547. https://doi.org/10.1111/gcb.13936
  Prentice, et al. The Inflection Point Hypothesis: The Relationship between the Temperature Dependence of Enzyme-Catalyzed Reaction Rates and Microbial Growth Rates. Biochemistry 2020, 59, 38, 3562–3569. https://doi.org/10.1021/acs.biochem.0c00530
  Alster, et al. Quantifying thermal adaptation of soil microbial respiration. Nat. Commun. 2023, 14, 5459. https://doi.org/10.1038/s41467-023-41096-x

2. And the another is a very recent model introduced by Arroyo et al. (2022) in PNAS. Their model provides a good fit to empirical data for a wide variety of biological rates, times, and steady-state quantities, from molecular to ecological scales, and it takes a simple form ln⁡y=ln⁡y0 - bx - a ln⁡x. (x = T^-1)
_Refs_:
  Arroyo, et al. A general theory for temperature dependence in biology. Proceedings of the National Academy of Sciences, 2022, 119(30): e2119872119. https://doi.org/10.1073/pnas.2119872119

**Comment by padpadpadpad** [comment]
Hi @Jinta0Li. Thanks for reaching out. Would you be willing to help add these models to the R package? If so I can advise on what information we would need.

Cheers
Dan

**Comment by Jinta0Li** [comment]
Yes, I would very much like to. Please tell me what I need to do.

Cheers,

Jintao

**Comment by padpadpadpad** [comment]
Nice!

rTPC is basically a set of helper functions to fit the most common rTPC models, but the fitting process is done using nls.multstart and other packages. Consequently if you just wrote your own function, it would like work with the "fitting many models" routine, apart from you would have to set your own start values and lower and upper limits. I think calc_params() should also work out the box as it just uses the model object to calculate the extra traits.

Things we would need are:

- a function for the model (named for the year of first use in the literature)
- know its sensible upper and lower limits (based either on the data or on parameter values from the literature) and these added to get_lower_lims() and get_upper_lims()
- know how to get sensible start values (based either on the data or on parameter values from the literature) and add these to get_start_vals()
- add the model name to get_model_names()

Happy to help where needed.

We could track progress in here. Do you have a function for either of the models?

**Comment by Jinta0Li** [comment]
Got it. While I'm not a code expert, I will try my best to consolidate what I know about these models and maybe send you what you need next week (Currently, I am doing a field sampling).

**Comment by Jinta0Li** [comment]
Hi @padpadpadpad ,

Thanks for your patience! I’ve completed the three models we discussed. They work well with both the chlorella_tpc dataset from rTPC and my own dataset, fitting all curves successfully.

I’ve attached the .R files here for you to review:

- arroyo_2022.R: Based on Arroyo et al. (2022)

- hobbs_2013.R: Based on Hobbs et al. (2013) (also known as “Macromolecular Rate Theory” or MMRT)

- prentice_2020.R: Based on Prentice et al. (2020) (a modified version of MMRT)

If you need any additional files or specific information about the models, please feel free to let me know, and I’ll be happy to provide it.

Cheers,
Jintao
[models_for_rTPC_Jintao.zip](https://github.com/user-attachments/files/17665572/models_for_rTPC_Jintao.zip)

**Comment by padpadpadpad** [comment]
Hi @Jinta0Li. @fwimp has been doing some amazing development of the R package adding a bunch of models so I will try incorporate these in as well.
